### DISCUSSION TITLE
Docker image for ARM

### DISCUSSION THREAD

**Discussion by axelrindle** [mysteriumnetwork/node#3380] [Show and tell]
Hello there!

I was not able to find a pre-built docker image for the Raspberry Pi 4B running on the 32-bit Raspbian version, so I built one myself. I took the following steps to do so:

1. Clone this repository

```bash
git clone https://github.com/mysteriumnetwork/node && cd node
```

2. Checkout the correct version

```bash
git checkout 0.46.1
```

3. Build the image

```bash
docker build . -f bin/docker/alpine/Dockerfile --build-arg BUILD_VERSION=0.46.1 -t axelrindle/mysterium-node:0.46.1
```

4. Run it

```bash
docker run -d --restart unless-stopped \
        --cap-add NET_ADMIN \
        -p 4449:4449 \
        -p 27005:27005/udp \
        -p 41920-42075:41920-42075/udp \
        -p 61920-62075:61920-62075/udp \
        --name myst \
        -v /path/to/mysterium_data:/var/lib/mysterium-node \
        axelrindle/mysterium-node:0.46.1 \
        --experiment-natpunching=false \
        --p2p.listen.ports=41920:42075 \
        service \
        --agreed-terms-and-conditions \
        --wireguard.listen.ports=61920:62075 \
        --openvpn.port=27005 openvpn,wireguard
```

My build environment is the following:

- OS: Raspbian GNU/Linux 10 (buster) with Linux 5.10.17-v7l+ armv7l
- Mysterium Node Version: 0.46.1
- Docker: 20.10.6

I also pushed my custom build to DockerHub, available here: https://hub.docker.com/r/axelrindle/mysterium-node

Let me know if there is anything that I've missed in the build process, otherwise: Happy building 😄

**Comment by ghost** [comment]
Can this not be included in the [documentation for Raspberry PI](https://docs.mysterium.network/node-runners/setup/raspberry-pi/)? Would be great if the build at DockerHub could be synchronized with releases.

**Reply by adinetech** [reply]
> 

Feel free to edit the source and PR :)

**Comment by ghost** [comment]
@axelrindle : Now, an updated push to Docker hub would be great since the 0.47.1 version is now too old to qualify the bounty.
